tensorflow_proto_derive/src/lib.rs

```diff
@@ -0,0 +1,72 @@
+extern crate proc_macro;
+
+use proc_macro::TokenStream;
+use syn::DeriveInput;
+
+#[proc_macro_derive(SerdeDefaultViable)]
+pub fn derive_it(input: TokenStream) -> TokenStream {
+    let ast = syn::parse_macro_input!(input as DeriveInput);
+    let name = ast.ident.clone();
+
+    match syn::Item::from(ast) {
+        syn::Item::Struct(_) => {
+            quote::quote! {
+                paste::paste! {
+                    #[derive(Default, Clone, serde::Deserialize, serde::Serialize)]
+                    #[serde(transparent, default)]
+                    struct [<Serde #name>] {
+                        inner: #name
```

Comment:
Probably need a test to ensure this is not mutually recursive.

Reply:
(I suspect that `serde(transparent)` might be calling `#name::deserialize(..)` which then calls `Serde#name::deserialize(..)` which then calls `#name::deserialize(..)`.

Reply:
Hm, now I'm not entirely sure how to delegate here.

Reply:
IOW, how do I tell `serde` to use the `inner` serde?

Reply:
I think you might want to replicate the fields of the original type within `Serde#name` type and then convert from one to another in the `deserialize` method.

Reply:
Ultimately ended up depending on upstream `prost` to pickup this commit: d9547f173246480e69120300f41c190257e0b63f so I could implement this whole thing as an attribute macro, greatly simplifying the implementation.